src/com/google/javascript/jscomp/CollapseProperties.java

```diff
@@ -696,6 +700,16 @@ private void collapseDeclarationOfNameAndDescendants(Name n, String alias) {
 
     // Handle this name first so that nested object literals get unrolled.
     if (n.canCollapse()) {
+
+      // Enum properties are always collapsed. Warn when @nocollapse
+      // is used on an enum property
+      if (n.docInfo != null && n.docInfo.isNoCollapse() &&
+          n.parent != null && n.parent.type == Name.Type.OBJECTLIT &&
+          n.parent.docInfo != null && n.parent.docInfo.hasEnumParameterType()) {
```

Comment:
here, I would create a helper method in the Name class, called isDescendantOfEnum, and it would check not only the parent, but all ancestors of the name.